src/pages/index.js

```diff
@@ -1,101 +1,245 @@
 }
 
 function renderCard(cardData) {
   const cardElement = createCard(cardData);
   section.addItem(cardElement);
 }
 
-function handleImageClick(name, link) {
-  popupImage.open(name, link);
+function handleImageClick(cardData) {
+  popupImage.open(cardData);
 }
 
-function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo(data);
-  popupEditForm.close();
+function handleAvatarSubmit(url) {
+  editAvatarPopup.renderLoading(true);
+  api
+    .updateAvatar(url)
+    .then((res) => {
+      userInfo.updateAvatar(res);
+      avatarForm.reset();
+      avatarFormValidator.resetValidation();
+      editAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => editAvatarPopup.renderLoading(false));
 }
 
-function handleAddCardSubmit(cardData) {
-  renderCard({ name: cardData.title, link: cardData.url });
-  cardForm.reset();
-  addFormValidator.disableButton();
-  popupEditForm.close();
+function handleProfileEditSubmit({ name, about }) {
+  editProfilePopup.renderLoading(true);
+  api
+    .updateProfileInfo(name, about)
+    .then((userData) => {
+      userInfo.setUserInfo(userData);
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      editProfilePopup.renderLoading(false);
+    });
 }
 
-// BUTTONS////
+function handleAddCardSubmit({ title, url }) {
+  addCardPopup.renderLoading(true);
+
+  api
+    .addCard({ name: title, link: url })
+    .then((cardData) => {
+      const cardElement = createCard(cardData);
+      section.addItem(cardElement);
+      addCardPopup.close();
+      cardForm.reset();
+      addFormValidator.resetValidation();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      addCardPopup.renderLoading(false);
+    });
+}
+
+const modalDelete = new PopupConfirmDelete("#modal__delete");
+modalDelete.setEventListeners();
+
+function handleDeleteSubmit(card) {
+  modalDelete.open();
+
+  modalDelete.setSubmit(() => {
+    modalDelete.renderLoading(true);
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        card.deleteCardFromDom();
+        modalDelete.close();
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => {
+        modalDelete.renderLoading(false);
+      });
+  });
+}
+
+function handleLikeClicks(card) {
+  if (!card.isLiked) {
+    api
+      .likeCard(card._id)
+      .then(() => {
+        card.setIsLiked(true);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  } else {
+    api
+      .disLikeCard(card._id)
+      .then(() => {
+        card.setIsLiked(false);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
+}
+
+api
+  .getUserInfo()
+  .then((data) => {
+    userInfo.updateAvatar(data);
+    userInfo.setUserInfo({
+      name: data.name,
+      about: data.about,
+    });
+  })
+  .catch((err) => {
+    console.log(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors